src/commons/eventsourcing/store.rs

```diff
@@ -356,19 +212,31 @@ where
     }
 
     /// Adds a new aggregate instance based on the init event.
-    pub fn add(&self, init: A::InitEvent) -> StoreResult<Arc<A>> {
-        let handle = init.handle().clone();
+    pub fn add(&self, cmd: A::InitCommand) -> StoreResult<Arc<A>> {
+        let handle = cmd.handle().clone();
 
         let agg_lock = self.locks.for_handle(handle.clone());
         let _write_lock = agg_lock.write();
 
-        self.store_event(&init)?;
+        let processed_command_builder =
+            StoredCommandBuilder::<A>::new(cmd.actor().to_string(), Time::now(), handle.clone(), 0, cmd.store());
 
-        let aggregate = A::init(init).map_err(|_| AggregateStoreError::InitError(handle.clone()))?;
-        self.store_snapshot(&handle, &aggregate)?;
+        let init_event = A::process_init_command(cmd).map_err(|_| AggregateStoreError::InitError(handle.clone()))?;
+        let aggregate = A::init(handle.clone(), init_event.clone());
+
+        // Store the init command. It is unlikely that this should fail, but
+        // if it does then there is an issue with the storage layer that we cannot
+        // recover from. So, exit.
+        let processed_command = processed_command_builder.finish_with_init_event(init_event);
+        if let Err(e) = self.store_command(&processed_command) {
+            error!("Cannot save state for '{}'. Got error: {}", handle, e);
+            error!("Will now exit Krill - please verify that the disk can be written to and is not full");
```

Comment:
As kvx is now (starting to be?) used is a reference to "disk" here still correct?

Reply:
I extracted this to a separate function and improved the error message.